### PR TITLE
Ignite APIs and several properties to ZCRMAttachment added

### PR DESCRIPTION
**Pull request by UmashriR**

### CODE REVIEW

File: ZCRMiOS/Classes/ModuleAPIHandler.swift [hunk 1]
```diff
@@ -447,6 +449,49 @@ internal class ModuleAPIHandler : CommonAPIHandler
...
+        addRequestParam(param: "module", value: self.module.getAPIName())
+        let request : APIRequest = APIRequest( handler: self )
+        print( "Request : \( request.toString() )" )
+        
+        request.getBulkAPIResponse { ( response, err ) in
+            if let error = err
+            {
+                completion( nil, nil, error )
+            }
+            if let bulkResponse = response
```
Comment (BoopathyPSiva) [inline ZCRMiOS/Classes/ModuleAPIHandler.swift]: use guard instead of if let.
Response nil isn't handled.

**Review by BoopathyPSiva — COMMENTED** [review]
Use guard instead of if let.
Response nil error isn't handled.

File: ZCRMiOS/Classes/MassEntityAPIHandler.swift [hunk 2]
```diff
@@ -422,27 +429,27 @@ internal class MassEntityAPIHandler : CommonAPIHandler
 	
 	// MARK: - Utility Functions
 	
-    internal func setTrashRecordProperties( record : [ String : Any ] )
+    private func setTrashRecordProperties( record : [ String : Any ] )
     {
         for ( fieldAPIName, value ) in record
         {
-            if( "created_By" == fieldAPIName )
+            if( ResponseParamKeys.deletedBy == fieldAPIName )
```
Comment (BoopathyPSiva) [inline ZCRMiOS/Classes/MassEntityAPIHandler.swift]: It should be ResponseParamKeys.createdBy

File: ZCRMiOS/Classes/MassEntityAPIHandler.swift [hunk 3]
```diff
@@ -422,27 +429,27 @@ internal class MassEntityAPIHandler : CommonAPIHandler
...
             {
                 let createdBy : [ String : Any ] = value as! [ String : Any ]
-                let createdByUser : ZCRMUser = ZCRMUser( userId : createdBy.getInt64( key : "id" ), userFullName : createdBy.getString( key : "name") )
+                let createdByUser : ZCRMUser = ZCRMUser( userId : createdBy.getInt64( key : ResponseParamKeys.id ), userFullName : createdBy.getString( key : ResponseParamKeys.name) )
                 self.trashRecord.setCreatedBy( createdBy : createdByUser )
             }
-            else if( "deleted_by" == fieldAPIName )
+            else if( ResponseParamKeys.createdBy == fieldAPIName )
```
Comment (BoopathyPSiva) [inline ZCRMiOS/Classes/MassEntityAPIHandler.swift]: It should be ResponseParamKeys. deletedBy

File: ZCRMiOS/Classes/TagAPIHandler.swift [hunk 4]
```diff
@@ -82,26 +83,31 @@ internal class TagAPIHandler : CommonAPIHandler
...
             {
                 completion( nil, ZCRMError.ProcessingError( "Tag ID MUST NOT be nil" ) )
+                return
             }
-            let tagIdString : String = String(tag.getId()!)
-            setJSONRootKey(key: JSONRootKey.TAGS)
-            setUrlPath(urlPath: "/settings/tags/\(tagIdString)/actions/records_count")
-            setRequestMethod(requestMethod: .GET)
-            addRequestParam(param: "module", value: module.getAPIName())
+            else
```
Comment (BoopathyPSiva) [inline ZCRMiOS/Classes/TagAPIHandler.swift]: remove else. else is not needed here

File: ZCRMiOS/Classes/TagAPIHandler.swift [hunk 5]
```diff
@@ -82,26 +83,31 @@ internal class TagAPIHandler : CommonAPIHandler
...
-                    let responseJSON = response.getResponseJSON()
-                    let count = Int64( responseJSON.getString( key : "count" ) )
-                    completion( count, nil )
+                request.getAPIResponse { ( resp, error ) in
+                    if let err = error
+                    {
+                        completion( nil, err )
+                        return
+                    }
+                    if let response = resp
```
Comment (BoopathyPSiva) [inline ZCRMiOS/Classes/TagAPIHandler.swift]: use guard here

File: ZCRMiOS/Classes/TagAPIHandler.swift [hunk 6]
```diff
@@ -179,12 +186,13 @@ internal class TagAPIHandler : CommonAPIHandler
...
             {
                 completion( nil, nil, ZCRMError.ProcessingError( "Tag ID MUST NOT be nil" ) )
+                return
             }
             else
             {
                 var conflictTagJSON : [String:Any] = self.getZCRMTagAsJSON(tag: withTag) as Any as! [String:Any]
                 var conflictIdJSON : [String:Any] = [String:Any]()
-                conflictIdJSON["conflict_id"] = conflictTagJSON["id"]
+                conflictIdJSON[RequestParamKeys.conflictId] = conflictTagJSON[ResponseParamKeys.id]
```
Comment (BoopathyPSiva) [inline ZCRMiOS/Classes/TagAPIHandler.swift]: remove else. else is not needed here

**Review by BoopathyPSiva — APPROVED** [review]

File: ZCRMiOS/Classes/ZCRMOrganisation.swift [hunk 7]
```diff
@@ -399,6 +399,20 @@ public class ZCRMOrganisation : ZCRMEntity
             completion( role, response, error )
         }
     }
+    
+    public func searchUserByCriteria( criteria : String, completion : @escaping( [ ZCRMUser ]?, BulkAPIResponse?, Error? ) -> () )
```
Comment (vijayakrishnap) [inline ZCRMiOS/Classes/ZCRMOrganisation.swift]: have method name as "searchUsers" and label for first argument as "withCriteria"

**Review by vijayakrishnap — CHANGES_REQUESTED** [review]

**Review by vijayakrishnap — APPROVED** [review]

**Pull request merged**